src/bun.js/api/bun/x509.zig

```diff
@@ -0,0 +1,553 @@
+        // exist for the same key. That is not great but there is not much we can
+        // change here without breaking things. Note that this creates nested data
+        // structures, yet still does not allow representing Distinguished Names
+        // accurately.
+        if (result.getTruthy(globalObject, name_slice)) |value| {
+            if (value.jsType().isArray()) {
+                value.push(globalObject, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+            } else {
+                const prop_name = JSC.ZigString.fromUTF8(name_slice);
+                const array = JSValue.createEmptyArray(globalObject, 2);
+                array.putIndex(globalObject, 0, value);
+                array.putIndex(globalObject, 1, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+                result.put(globalObject, &prop_name, array);
+            }
+        } else {
+            const prop_name = JSC.ZigString.fromUTF8(name_slice);
+            result.put(globalObject, &prop_name, JSC.ZigString.fromUTF8(value_slice).toValueGC(globalObject));
+        }
+    }
+    return result;
+}
+
+inline fn isSafeAltName(name: [*c]const u8, length: usize, utf8: bool) bool {
+    for (0..length) |i| {
+        const c = name[i];
+
+        switch (c) {
+            '"',
+            '\\',
+            // These mess with encoding rules.
+            // Fall through.
+            ',',
+            // Commas make it impossible to split the list of subject alternative
+            // names unambiguously, which is why we have to escape.
+            // Fall through.
+            '\'',
+            => {
+                // Single quotes are unlikely to appear in any legitimate values, but they
+                // could be used to make a value look like it was escaped (i.e., enclosed
+                // in single/double quotes).
+                return false;
+            },
+            else => {
+                if (utf8) {
+                    // In UTF8 strings, we require escaping for any ASCII control character,
+                    // but NOT for non-ASCII characters. Note that all bytes of any code
+                    // point that consists of more than a single byte have their MSB set.
+                    if (c < ' ' or c == '\x7f') {
+                        return false;
+                    }
+                } else {
+                    // Check if the char is a control character or non-ASCII character. Note
+                    // that char may or may not be a signed type. Regardless, non-ASCII
+                    // values will always be outside of this range.
+                    if (c < ' ' or c > '~') {
+                        return false;
+                    }
+                }
+            },
+        }
+    }
+    return true;
+}
+
+inline fn printAltName(out: *BoringSSL.BIO, name: [*]const u8, length: usize, utf8: bool, safe_prefix: ?[*]const u8) void {
```

Comment:
these should be slices

Reply:
fix on https://github.com/oven-sh/bun/commit/927f54269bdf489d21be4a8a6b89b448fdf48184